pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -2398,11 +2398,27 @@ func (k kafkaManager) configureHydraOauth(ctx context.Context) (string, string)
 		err := k.c.RunE(ctx, option.WithNodes(k.kafkaSinkNode), `/home/ubuntu/hydra-serve.sh`)
 		return errors.Wrap(err, "hydra failed")
 	})
-	result, err := k.c.RunWithDetailsSingleNode(ctx, k.t.L(), option.WithNodes(k.kafkaSinkNode), "/home/ubuntu/hydra create oauth2-client",
-		"-e", "http://localhost:4445",
-		"--grant-type", "client_credentials",
-		"--token-endpoint-auth-method", "client_secret_basic",
-		"--name", `"Test Client"`,
+
+	var result install.RunResultDetails
+	// The admin server may not be up immediately, so retry the create command
+	// until it succeeds or times out.
+	err = timeutil.RunWithTimeout(
```

Comment:
I think the `retry` package is a better fit for this: it will automatically apply some backoff instead of trying to run the command non-stop like is being done here (which `sshd` might not like).